### DISCUSSION TITLE
Idea: BOLT12

### DISCUSSION THREAD

**Discussion by matthiasdebernardini** [fedimint/fedimint#1507] [Fedimint Modules]
Fork existing lightning module and make it support BOLT12

**Comment by elsirion** [comment]
This should be easy once we have a stable FROST/ROAST implementation that we can use in Fedimint to threshold-sign generated invoices.

EDIT: I created a flow chart how BOLT12 support could work
![fedimin_multi_invoices drawio](https://user-images.githubusercontent.com/84478054/222920945-9f5de888-37b1-4b93-97e8-77364c24bdf1.png)

**Reply by otech47** [reply]
Do we also need APIs for querying how many preimages for a given offer id are available to use so the user can topup and add more preimages for the offer?

**Reply by elsirion** [reply]
Yeah, we'd need that, but I have to think about how this flow would work with LNv2 since the Federation is less involved there and more of the interaction is between the user and the gateway.
